examples/migrationExample/src/main/java/io/realm/examples/realmmigrationexample/model/Migration.java

```diff
@@ -45,32 +50,32 @@ public long execute(Realm realm, long version) {
             // Version 1
             class Person
                 @Required
-                String fullName;        // combine firstName and lastName into single field.
+                String fullName;            // combine firstName and lastName into single field.
                 int age;
-        */
-
+        ************************************************/
         // Migrate from version 0 to version 1
-        if (version == 0) {
-            Table personTable = realm.getTable(Person.class);
-
-            long fistNameIndex = getIndexForProperty(personTable, "firstName");
-            long lastNameIndex = getIndexForProperty(personTable, "lastName");
-            long fullNameIndex = personTable.addColumn(RealmFieldType.STRING, "fullName");
-            for (int i = 0; i < personTable.size(); i++) {
-                personTable.setString(fullNameIndex, i, personTable.getString(fistNameIndex, i) + " " +
-                        personTable.getString(lastNameIndex, i));
-            }
-            personTable.removeColumn(getIndexForProperty(personTable, "firstName"));
-            personTable.removeColumn(getIndexForProperty(personTable, "lastName"));
-            version++;
+        if (oldVersion == 0) {
+            RealmObjectSchema personSchema = schema.get("Person");
+
+            // Combine 'firstName' and 'lastName' in a new field called 'fullName'
+            personSchema
+                    .addField("fullName", String.class, FieldAttribute.REQUIRED)
+                    .forEach(new RealmObjectSchema.Transformer() {
+                        @Override
+                        public void apply(DynamicRealmObject obj) {
+                            obj.set("fullName", obj.getString("firstName") + " " + obj.getString("lastName"));
```

Comment:
This might be performant, but I find the API really unpleasant. As we discussed previously I would much rather have something that at least returns a `DynamicRealmObject ` but ideally also has two different objects, one for input and one for output